### ISSUE TITLE
Refactor "swap-if-else-branches" breaks code with inner function

### ISSUE THREAD

**Issue by dimitri-b** [sourcery-ai/sourcery#193]
### Applying suggested refactoring to a decorator function deletes the outer function code

See attached example with code.

### Sourcery Version

0.10.2

### Code editor or IDE name and version

Pycharm 2021.3.2 Pro

### OS name and version

OSX 11.6.1

### Example code
```python
# suggested refactoring: swap-if-else-branches


# ------------- before refactoring --------------

def retry(func: Callable) -> Callable:
    """Decorator: retry with exp. backoff until max_wait is reached."""
    wait_cap = 10
    max_wait = 30
    start = time.time()

    def wrapper(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except Exception as err:
            elapsed = time.time() - start
            if elapsed < max_wait:
                wrapper.attempt_count += 1
                delay = 2 ** wrapper.attempt_count
                time.sleep(min(delay, wait_cap))
                wrapper(*args, **kwargs)
            else:
                raise err

    wrapper.attempt_count = 0
    return wrapper


# -------------- after refactoring --------------

def wrapper(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except Exception as err:
            elapsed = time.time() - start
            if elapsed >= max_wait:
                raise err
            wrapper.attempt_count += 1
            delay = 2**wrapper.attempt_count
            time.sleep(min(delay, wait_cap))
            wrapper(*args, **kwargs)



```

[sourcery_retry_decorator_refactor_issue.txt](https://github.com/sourcery-ai/sourcery/files/8065088/sourcery_retry_decorator_refactor_issue.txt)

**Comment by brendanator** [comment]
Thanks for reporting this 🙏

I have reproduced and am investigating

**Comment by brendanator** [comment]
We have a fix for this which be part of the next release

**Comment by Hellebore** [comment]
Sadly this didn't quite make it into 0.10.3 - but will be in the next one.

**Comment by Hellebore** [comment]
Fixed with release of 0.11.1
